astropy/units/tests/test_format.py

```diff
@@ -425,7 +406,7 @@ class TestRoundtripOGIP(RoundtripBase):
             for unit in u_format.OGIP._units.values()
             if (isinstance(unit, UnitBase) and not isinstance(unit, PrefixUnit))
```

Comment:
The `_units` property is supposed to be a `dict[str, UnitBase]`, so checking `isinstance(unit, UnitBase)` should not be required. The most interesting outcome of this little exercise in shortening code is that for `OGIP` the check is in fact still required. The reason is that `OGIP` (and only `OGIP`) inserts functions to `_units`: https://github.com/astropy/astropy/blob/15f56205693a3e0220907aee4a1618248c500149/astropy/units/format/ogip.py#L98
This is different from `Generic`, which lets the tokenizer recognize functions https://github.com/astropy/astropy/blob/15f56205693a3e0220907aee4a1618248c500149/astropy/units/format/generic.py#L109-L111
and has separate parser rules for handling them: https://github.com/astropy/astropy/blob/15f56205693a3e0220907aee4a1618248c500149/astropy/units/format/generic.py#L418-L420

I am not aware of any good reasons why `OGIP` should differ from the other parsing formatters in this regard. Differences like this make our unit parser harder to understand, so it is worth investigating if it would be possible to rewrite this part of `OGIP` to be more like the other formatters.

I've been running `mypy` on the unit formatters, but this type violation in `OGIP._units` has gone unnoticed so far because `mypy` gets confused by our `classproperty` decorator.

Reply:
Just out of curiosity, is my pipe passing on the unit formatting section? Is it worth turning mypy on now for this part of the package?

Reply:
Agreed that `OGIP` should logically not have the functions in the names.

Reply:
> Just out of curiosity, is mypy passing on the unit formatting section? Is it worth turning mypy on now for this part of the package?

Short answer: no